src/main/java/fr/esgi/al5_2/Tayarim/auth/JwtHelper.java

```diff
@@ -18,75 +18,96 @@
+   */
+  public String generateToken(String email, String uuid, boolean isAdmin) {
+    String subject = email.concat(";").concat(uuid).concat(";").concat(Boolean.toString(isAdmin));
+    var now = Instant.now();
+    return Jwts.builder()
+        .subject(subject)
+        .issuedAt(Date.from(now))
+        .expiration(Date.from(now.plusSeconds(86400))) //24h
+        .signWith(getSignKey())
+        .compact();
+  }
 
-    public String extractEmail(String token) {
-        return extractSubject(token).split(";")[0];
+  public String extractSubject(String token) {
+    try {
+      Claims tokenBody = getTokenBody(token);
+      if (tokenBody == null) {
+        throw new TokenExpireOrInvalidException();
+      }
+
+      return tokenBody.getSubject();
+    } catch (Exception e) {
+      throw new TokenExpireOrInvalidException();
     }
+  }
 
-    public String extractUuid(String token) {
-        return extractSubject(token).split(";")[1];
+  public String extractEmail(String token) {
+    if (extractSubject(token).split(";")[0] == null) {
+      throw new TokenExpireOrInvalidException();
     }
 
-    public Boolean validateToken(String token, String email, String uuid) {
-        String extractedEmail = extractEmail(token);
-        String extractedUUID = extractUuid(token);
+    return extractSubject(token).split(";")[0];
+  }
 
-        return extractedEmail != null &&
-                extractedUUID != null &&
-                extractedEmail.equals(email) &&
-                extractedUUID.equals(uuid) &&
-                !isTokenExpired(token);
+  public String extractUuid(String token) {
+    if (extractSubject(token).split(";")[1] == null) {
+      throw new TokenExpireOrInvalidException();
     }
 
-    private Claims getTokenBody(String token) {
-        try {
-            return Jwts
-                    .parser()
-                    .verifyWith(getSignKey())
-                    .build()
-                    .parseSignedClaims(token)
-                    .getPayload();
-        } catch (SignatureException | ExpiredJwtException e) { // Invalid signature or expired token
-            return null;
-        }
+    return extractSubject(token).split(";")[1];
+  }
+
+  public Boolean extractAdmin(String token) {
+    if (extractSubject(token).split(";")[2] == null) {
+      throw new TokenExpireOrInvalidException();
     }
 
-    private boolean isTokenExpired(String token) {
-        Claims claims = getTokenBody(token);
-        if (claims == null){
-            return true;
-        }
-        return claims.getExpiration().before(new Date());
+    return Boolean.parseBoolean(extractSubject(token).split(";")[2]);
+  }
+
+  public Boolean validateToken(String token, String email, String uuid) {
+    String extractedEmail = extractEmail(token);
+    String extractedUUID = extractUuid(token);
+
+    return extractedEmail != null &&
+        extractedUUID != null &&
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.OperatorWrapCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'&&' should be on a new line.